src/vs/platform/terminal/common/capabilities/commandDetection/terminalCommand.ts

```diff
@@ -0,0 +1,390 @@
+				lines.push(getXtermLineContent(buffer, wrappedLineStart, wrappedLineEnd, this._xterm.cols));
+				if (!match) {
+					match = lines[lines.length - 1].match(matcher);
+				}
+				if (lines.length >= linesToCheck) {
+					break;
+				}
+			}
+		}
+		return match ? { regexMatch: match, outputLines: lines } : undefined;
+	}
+
+	hasOutput(): boolean {
+		return (
+			!this.executedMarker?.isDisposed &&
+			!this.endMarker?.isDisposed &&
+			!!(
+				this.executedMarker &&
+				this.endMarker &&
+				this.executedMarker.line < this.endMarker.line
+			)
+		);
+	}
+
+	getPromptRowCount(): number {
+		return getPromptRowCount(this, this._xterm.buffer.active);
+	}
+
+	getCommandRowCount(): number {
+		return getCommandRowCount(this);
+	}
+}
+
+export interface ICurrentPartialCommand {
+	previousCommandMarker?: IMarker;
+
+	promptStartMarker?: IMarker;
+
+	commandStartMarker?: IMarker;
+	commandStartX?: number;
+	commandStartLineContent?: string;
+
+	commandRightPromptStartX?: number;
+	commandRightPromptEndX?: number;
+
+	commandLines?: IMarker;
+
+	commandExecutedMarker?: IMarker;
+	commandExecutedX?: number;
+
+	commandFinishedMarker?: IMarker;
+
+	currentContinuationMarker?: IMarker;
+	continuations?: { marker: IMarker; end: number }[];
+
+	command?: string;
+
+	/**
+	 * Whether the command line is trusted via a nonce.
+	 */
+	isTrusted?: boolean;
+
+	/**
+	 * Something invalidated the command before it finished, this will prevent the onCommandFinished
+	 * event from firing.
+	 */
+	isInvalid?: boolean;
+
+	getPromptRowCount(): number;
+	getCommandRowCount(): number;
+}
+
+export class PartialTerminalCommand implements ICurrentPartialCommand {
+	previousCommandMarker?: IMarker;
+
+	promptStartMarker?: IMarker;
+
+	commandStartMarker?: IMarker;
+	commandStartX?: number;
+	commandStartLineContent?: string;
+
+	commandRightPromptStartX?: number;
+	commandRightPromptEndX?: number;
+
+	commandLines?: IMarker;
+
+	commandExecutedMarker?: IMarker;
+	commandExecutedX?: number;
+
+	commandFinishedMarker?: IMarker;
+
+	currentContinuationMarker?: IMarker;
+	continuations?: { marker: IMarker; end: number }[];
+
+	command?: string;
+
+	isTrusted?: boolean;
+	isInvalid?: boolean;
+
+	constructor(
+		private readonly _xterm: Terminal,
+	) {
+	}
+
+	serialize(cwd: string | undefined): ISerializedTerminalCommand | undefined {
+		if (!this.commandStartMarker) {
+			return undefined;
+		}
+
+		return {
+			promptStartLine: this.promptStartMarker?.line,
+			startLine: this.commandStartMarker.line,
+			startX: this.commandStartX,
+			endLine: undefined,
+			executedLine: undefined,
+			executedX: undefined,
+			command: '',
+			isTrusted: true,
+			cwd,
+			exitCode: undefined,
+			commandStartLineContent: undefined,
+			timestamp: 0,
+			markProperties: undefined
+		};
+	}
+
+	promoteToFullCommand(cwd: string | undefined, exitCode: number | undefined, ignoreCommandLine: boolean, markProperties: IMarkProperties | undefined): TerminalCommand | undefined {
```

Comment:
`resolveAsFullCommand` ?

Reply:
It was convert initially, I think promote does as good job of explaining it